docs/guides/scheduler_event.md

```diff
@@ -0,0 +1,107 @@
+# The scheduler event
+
+The scheduler event from Concord is the best way to get the RAW JSON that Discord sends to Concords, with it you can allow doing a lot of things, for example, you can make Concord ignore an event, or even better, just parse a JSON key manually to achieve the best performance from it.
+
+## How to use it
+
+To usage of the scheduler is easy, take a look at the example below of printing the raw JSON from Discord of the MESSAGE_CREATE event and making Concord ignore the event:
+
+```c
+enum discord_event_scheduler scheduler(struct discord *client, const char data[], size_t length, enum discord_gateway_events event) {
+    /* The "event" parameter is the parameter allows you to distinguish the events, and the data one is the JSON Discord sent. */
+    if (event == DISCORD_EV_MESSAGE_CREATE) {
+        log_debug("Received event JSON: %s", data);
+        /* Returning the "event ignore" enum, that says to Concord to ignore the event. */
+        return DISCORD_EVENT_IGNORE;
+    }
+    /* Here we are "saying" to Concord to handle the rest of the received events. */
+    return DISCORD_EVENT_MAIN_THREAD;
+    /* The "length" parameter can be used to parse the JSON, you can use jsmnf, from müller, in case you want to parse separated fields, or you can use discord_xxx_from_json() functions, in case you want to parse the whole JSON. */
+}
+```
+
+And this is it, but don't forget to add `discord_set_event_scheduler(client, scheduler);` to your code, otherwise, Concord will not use the scheduler.
+
+## How to parse the JSON (discord_xxx_from_json)
+
+This method is by far the easiest one, but it is not the fastest one, and you will need to free the returned struct, but this won't be a problem for most cases, so don't worry.
+
+`discord_xxx_from_json` isn't a single function, it's a set of functions, one for each event, so you will need to use the correct one, for example, if you want to parse a `message create` event, you will need to use `discord_message_from_json`, but never forget to free the returned struct, otherwise you will have a memory leak.
+
+Take an example of parsing a `message create` event using scheduler:
+
+```c
+enum discord_event_scheduler scheduler(struct discord *client, const char data[], size_t length, enum discord_gateway_events event) {
+    if (event == DISCORD_EV_MESSAGE_CREATE) {
+        struct discord_message self;
+        /* Parse the JSON and store the it in the "self" struct. */
+        discord_message_from_json(data, length, &self);
+
+        log_debug("Message content: %s", self.content);
+
+        /* Free the struct. */
+        discord_message_cleanup(&self);
+        return DISCORD_EVENT_IGNORE;
+    }
+    return DISCORD_EVENT_MAIN_THREAD;
+}
+```
```

Comment:
```suggestion
Scheduler also allows the use of `discord_xxx_from_json()` to convert the JSON into a struct where you can then produce the logic you need. Here, `xxx` implies a family of functions. In this case, the `xxx` in `discord_xxx_from_json` can be replaced with events like `message_create`.

Here is an example of using the Scheduler to ignore the `MESSAGE_CREATE` event and print out its JSON:
```c
enum discord_event_scheduler scheduler(struct discord *client, const char data[], size_t length, enum discord_gateway_events event) {
    if (event == DISCORD_EV_MESSAGE_CREATE) {
        struct discord_message self;
        /* Parse the JSON and store the it in the "self" struct. */
        discord_message_from_json(data, length, &self);

        log_debug("Message content: %s", self.content);

        /* Free the struct. */
        discord_message_cleanup(&self);
        return DISCORD_EVENT_IGNORE;
    }
    return DISCORD_EVENT_MAIN_THREAD;
}
```
```
Make sure to keep the main idea first, and then make notes about it such as freeing the structure.

Reply:
Sorry, didn't understand, do you have remove the part saying the cons and pros of the section's method or move them to the final of the section?